Fix session sigs

hwrdtm opened the pull request:
# What

Closes LIT-294.

# TODO

TODOs before being reviewable:

- [x] Test against local rust node, local JS SDK, local relay server, local contracts.
- [x] Use published JS SDK version.

hwrdtm committed:
Decouple Google OAuth sign in with Mint PKP step

hwrdtm committed:
Use new signSessionKey SDK method

hwrdtm committed:
Handle authentication separately from storing conditions

hwrdtm committed:
Use auth-helpers package to generate session sigs correctly

hwrdtm committed:
Use react-app-rewired to use config override for fixing webpack bundle

create-react-app default webpack config does not match against .cjs files, so we need to override the default config to bundle .cjs files from auth-helpers using CJS to import the .cjs file
from recap-ts.

hwrdtm committed:
Implement encrypt / decrypt using session sigs

johnnynanjiang commented:
Hi @hwrdtm, when would the TODOs be completed and review be done? Thanks.

joshLong145 commented on src/App.tsx:
```diff
@@ -234,9 +285,37 @@ function App() {
+	}) => void
+) => {
+	setStatusFn("Minting PKP...");
+	const requestId = await mintPkpUsingRelayerGoogleAuthVerificationEndpoint(
+		credentialResponse,
+		setStatusFn
+	);
+	return pollRequestUntilTerminalState(requestId, setStatusFn, onSuccess);
```
Does this use the `lit-auth-client`?

hwrdtm replied:
No but we can do, in a future PR. I'd like us to get session sigs working first, then use the higher-level abstraction later. But @johnnynanjiang by all means check out our newest `lit-auth-client` if you're interested! https://github.com/LIT-Protocol/js-sdk/tree/master/packages/lit-auth-client

hwrdtm committed:
Use latest js-sdk, use npm instead of yarn

yarn still installs tslib at 1.14.1, which breaks, whereas npm installs at 2.5.0
